src/analysis/retail/paladin/holy/modules/talents/LightsProtection.tsx

```diff
@@ -0,0 +1,55 @@
+import SPELLS from 'common/SPELLS';
+import { TALENTS_PALADIN } from 'common/TALENTS';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, { DamageEvent } from 'parser/core/Events';
+import Statistic from 'parser/ui/Statistic';
+import { ReactNode } from 'react';
+import TalentSpellText from 'parser/ui/TalentSpellText';
+import { formatNumber } from 'common/format';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+import { LIGHTS_PROTECTION_DAMAGE_REDUCTION } from '../../constants';
+
+class LightsProtection extends Analyzer {
+  totalDamagePrevented = 0;
+
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.hasTalent(TALENTS_PALADIN.LIGHTS_PROTECTION_TALENT);
+
+    this.addEventListener(Events.damage.to(SELECTED_PLAYER), this._onDamageTaken);
+  }
+
+  _combatantHasBeacon() {
+    return this.selectedCombatant.hasBuff(SPELLS.BEACON_OF_LIGHT_CAST_AND_BUFF.id);
+  }
+
+  _onDamageTaken(event: DamageEvent) {
+    if (!this._combatantHasBeacon()) {
+      this.totalDamagePrevented +=
+        (LIGHTS_PROTECTION_DAMAGE_REDUCTION * event.amount) /
+        (1 - LIGHTS_PROTECTION_DAMAGE_REDUCTION);
+    }
+  }
+
+  get DRPS() {
+    return (this.totalDamagePrevented / this.owner.fightDuration) * 1000;
+  }
+
+  statistic(): ReactNode {
+    return (
+      <Statistic
+        size="flexible"
+        category={STATISTIC_CATEGORY.TALENTS}
+        tooltip={<p>Estimated damage reduced: {formatNumber(this.totalDamagePrevented)}</p>}
+      >
+        <TalentSpellText talent={TALENTS_PALADIN.LIGHTS_PROTECTION_TALENT}>
+          <div>
+            {formatNumber(this.DRPS)} <small>estimated DRPS</small>
+          </div>
+        </TalentSpellText>
```

Comment:
this is only including damage reduction on yourself, but the talent reads "Allies with Beacon of Light receive 5% less damage". is it actually player-only or should this be adjusted (either text adjustment to clarify, or code adjustment to include DR on other players)

Reply:
mb, i have to fix it it's all players w/ beacon indeed

Reply:
is fetching WCL for additional data the only way to go ? Just checking out before I do something extremely inefficient

Reply:
we probably don't have helpers that you can use to determine who has the buff (though you might be able to iterate over friendlies?), but:

- we have `combatantinfo` events for every player, which *should* include pre-pull buffs. (if beacon is missing lmk and i'll pass it along)
- you can see `applybuff`/`removebuff` to other players by the paladin during combat, so if they move it you can track it
- worst case: you can see who is being healed by it and use that to determine who had the buff on pull

Reply:
https://github.com/WoWAnalyzer/WoWAnalyzer/blob/c3be75fbf6adc6cd6b30a19486847c5f8ee41645/src/parser/shared/modules/Combatants.ts#L12 this might be useful to iterate over players

Reply:
mmh maybe I'm just not seeing it but that seems a little expensive is the person is running Beacon of Virtue (the aoe beacon on a 15 sec CD), doesn't it ?

Reply:
oh i see the issue---you can tell when someone has beacon but not what damage they take.

yea this should be a load-on-demand stat. the preservation `TimeOfNeed` module does something similar to fetch the data, so you could start there (though that talent is a wee bit more complicated in how it works so everything *after* data loading is going to be different)

it is also okay for some talents to not have stats, even if interesting.

Reply:
Thanks for the answer, it looks cleaner than Lenience. There isn't any non-deprecated load-on-demand component ? 
Either way your last point might conclude this. I may go back to this later but there are more important talents to implement rn (hello hero talents)